packages/massa-proto-cli/src/index.ts

```diff
@@ -0,0 +1,135 @@
+import { MassaProtoFile } from '@massalabs/massa-web3/dist/esm/interfaces/MassaProtoFile';
+import { generateAsCallers } from './as-gen';
+import { ProtoFile } from './protobuf';
+import {
+  Client,
+  ClientFactory,
+  IProvider,
+  ProviderType,
+  SmartContractsClient,
+  WalletClient,
+} from '@massalabs/massa-web3';
+import * as dotenv from 'dotenv';
+
+// Load .env file content into process.env
+dotenv.config();
+
+// Get the URL for a public JSON RPC API endpoint from the environment variables
+const publicApi = process.env.JSON_RPC_URL_PUBLIC;
+if (!publicApi) {
+  throw new Error('Missing JSON_RPC_URL_PUBLIC in .env file');
+}
+
+// Get the secret key for the wallet to be used for the deployment from the environment variables
+const secretKey = process.env.WALLET_SECRET_KEY;
+if (!secretKey) {
+  throw new Error('Missing WALLET_SECRET_KEY in .env file');
+}
+
+// Create an account using the private key
+const deployerAccount = await WalletClient.getAccountFromSecretKey(secretKey);
+
+/**
+ * Displays the usage of the proto cli in terminal.
+ */
+function displayHelp() {
+  console.log('Usage:');
+  console.log('npx massa-proto-cli');
+  console.log(
+    '\t\t--gen sc|web3\t- the generation mode for contracts callers (sc) or web3 app (web3)',
+  );
+  console.log(
+    '\t\t--addr contract_public_key\t- the public address of the contract to interact with',
+  );
+  console.log(
+    '\t\t[--out output_directory]\t- optional output directory for the callers to generate (default: .)',
+  );
+  process.exit(1);
+}
+
+/**
+ * Fetching generation mode cli argument.
+ *
+ * @param args - arguments array
+ *
+ * @returns the generation mode
+ */
+function getGenMode(args: string[]): string {
+  let index = args.indexOf('--gen');
+  if (index >= 0 && index + 1 < args.length) {
+    return args[index + 1];
+  }
+  displayHelp();
+  return '';
+}
+
+/**
+ * Fetching address cli argument.
+ *
+ * @param args - arguments array
+ *
+ * @returns the address.
+ */
+function getAddr(args: string[]): string {
+  let index = args.indexOf('--addr');
+  if (index >= 0 && index + 1 < args.length) {
+    return args[index + 1];
+  }
+  displayHelp();
+  return '';
+}
+
+/**
+ * Fetching optionnal output directory cli argument.
+ *
+ * @param args - arguments array.
+ *
+ * @returns the output directory path.
+ */
+function getDir(args: string[]): string {
+  let index = args.indexOf('--out');
+  if (index >= 0 && index + 1 < args.length) {
+    return args[index + 1];
+  }
+  return './helpers/';
+}
```

Comment:
Why not using a library specialized on this kind of things ?

Reply:
Alaways learned to do this by my self (very scolar my bad), if you have a well known package to do that job i'm open to it :+1:

Reply:
:man_shrugging: